addons/mrp/models/res_config_settings.py

```diff
@@ -34,7 +34,11 @@ def _onchange_group_mrp_routings(self):
         # Work Orders' is deactivated.
         # Long story short: if 'mrp_workorder' is already installed, we don't uninstall it based on
         # group_mrp_routings
+        operations = self.env['mrp.routing.workcenter'].with_context({'active_test': False}).search([('company_id', '=', self.company_id.id)])
         if self.group_mrp_routings:
             self.module_mrp_workorder = True
-        elif not self.env['ir.module.module'].search([('name', '=', 'mrp_workorder'), ('state', '=', 'installed')]):
-            self.module_mrp_workorder = False
+            operations.action_unarchive()
```

Comment:
I wouldn't do that since if he archived operation before unistall module, it would reactivate it again. I would just let everything archive and let the user unarchive what he wants.